### ISSUE TITLE
Add Colorable Backpacks

### ISSUE THREAD

**Issue by bigtruckin78** [gr8pefish/IronBackpacks#61]
Hi,
I think a color upgrade option would be nice so it would be easier to differentiate bags at a glance.

Loving the mod, its my favorite backpack mod I've used. 
Thanks for the awesome work. :D

**Comment by gr8pefish** [comment]
Yup, that's already planned for the 1.8 version

**Comment by TheTerrorByte** [comment]
1.7 please? Not everyone can use 1.8 for mods you know.

**Comment by mcdodge34** [comment]
Bringing this back for attention, that would be very welcome, now that I have multiple bags with me, even with a different name, they look all the same...

**Comment by gr8pefish** [comment]
I'm actually working on this right now :)

I'll close this issue once I finish it.

On Mar 18, 2017 12:29 PM, "mcdodge34" <notifications@github.com> wrote:

Bringing this back for attention, that would be very welcome, now that I
have multiple bags with me, even with a different name, they look all the
same...

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/gr8pefish/IronBackpacks/issues/61#issuecomment-287568921>,
or mute the thread

**Comment by mcdodge34** [comment]
Cool, good to see another mod dev that reply to the requests, will wait impatiently for this feature like a 3-4yrs old that waits for the bathroom door...

**Comment by DragGon7601** [comment]
I came looking to see if this was already a feature. I disappointed to find out that its both not and likely will not be added to the 1.7.10 version that I play. Some major mods I play are not updated because they are very large and would basically need a complete rewrite. 

Once you add it to 1.8 I request that you look at back porting the feature; If its just going to be a 5-10 min job I ask that you try. I don't expect you to spend hours on it.

This would be much appreciated.

**Comment by mcdodge34** [comment]
Any news on this one, any hope to see this in 1.10.2 or your focus is more on 1.11.2 and 1.12 ??? Just curious here.

**Comment by gr8pefish** [comment]
Hey, sorry for the lack of a response.

I may put it on 1.10, but chances are slim for that. I tried to make it work (hence my optimistic message back in March), but I was hung up from some technical difficulties (e.g. I managed to allow coloring of _all_ backpacks of a type, but it wasn't specific to each backpack). Unfortunately, this was partially a result of the code debt I had built up over the years, so unless I get a large chunk of time I'm going to leave the 1.10 version as-is. 

I am addressing this via a rewrite for 1.11/1.12, and you can definitely expect the 1.12 version to have coloring though!

If you want to check out the progress of that port, check out the master/default branch (I do unfortunately jump branches a bit, but will eventually merge everything into that one.) I plan on releasing a quick, unsupported 1.11 version, and then quickly moving on to 1.12. 

TL;DR: It's taking a fair bit of time for this rewrite, so I have to focus my attention on future versions so as to maximize the usefulness of my limited time. Therefore treat 1.10 as feature complete, but expect coloring (and more hopefully) for 1.12.

**Comment by mcdodge34** [comment]
Ok, good to know its still in your plans, still use the backpacks anyway, nothing can beat them in terms of functionality. One thing that could be interesting would be some kind of a link to a ender storage container, so lets say your mining, the stuff go in the backpack and is transferred to a ender chest that is hooked to your ore processing plan, you never ever have to worry about storage space anymore.

**Comment by gr8pefish** [comment]
>still use the backpacks anyway, nothing can beat them in terms of functionality

:D

> link to a ender storage container

Yeah, ender functionality is planned, but I'm still tossing around ideas of how best to implement it. If you want to provide suggestions I'd love to hear them, but don't do it in this thread (keep this for colors); feel free to make a new issue though.

**Comment by gr8pefish** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Colorable Backpacks](https://trello.com/c/IUqzVhMo/9-colorable-backpacks)

**Comment by gr8pefish** [comment]
2 years later -> ✔️

**Comment by gr8pefish** [comment]
Implemented in the latest 1.12 release
